src/firestore/app/run.ts

```diff
@@ -173,30 +203,135 @@ export class RoarRun {
    * @async
    * @param {*} trialData - An object containing trial data.
    */
-  async writeTrial(trialData: Record<string, unknown>) {
+  async writeTrial(
+    trialData: Record<string, unknown>,
+    computedScoreCallback?: (rawScores: IRawScores) => IComputedOrNormedScores,
```

Comment:
is this where we can modify transform the raw score to computed score? how will the set up looks like in the app side?

Reply:
Yes, exactly. It will have to be passed in as an argument to `writeTrial`. For example, in the current SWR, you do this [here](https://github.com/yeatmanlab/roar-swr/blob/9a6d9cb9dbf4b2d7eefa00c7005d2480e58fff04/src/experiment.js#L328), but I think the location will change with Phil's refactor.

We can also ask Phil to change the template to allow the researcher to put it somewhere else in their code.

Reply:
ok! so it will be a function passed in writeTrial?